android-lib/src/androidTest/java/com/kinvey/androidTest/store/data/DataStoreMultiInsertTest.java

```diff
@@ -0,0 +1,1954 @@
+    private List<Person> createErrList2() {
+        List<Person> items = new ArrayList<>();
+        Person person1 = new Person(TEST_USERNAME + String.valueOf(1));
+        items.add(person1);
+        Person person2 = new Person(TEST_USERNAME + String.valueOf(2));
+        person2.setGeoloc(ERR_GEOLOC);
+        items.add(person2);
+        Person person3 = new Person(TEST_USERNAME + String.valueOf(3));
+        items.add(person3);
+        return items;
+    }
+
+    // create an array  - [{no_id, invalid_geoloc},{_id}, {_id, invalid_geoloc}, {no_id}]
+    private List<Person> createErrListGeoloc() {
+        List<Person> items = new ArrayList<>();
+        String errStr = ERR_GEOLOC;
+        Person person1 = new Person(TEST_USERNAME + String.valueOf(1));
+        person1.setGeoloc(errStr);
+        items.add(person1);
+        Person person2 = new Person("76575", TEST_USERNAME + String.valueOf(2));
+        items.add(person2);
+        Person person3 = new Person("343275", TEST_USERNAME + String.valueOf(3));
+        person1.setGeoloc(errStr);
+        items.add(person3);
+        Person person4 = new Person(TEST_USERNAME + String.valueOf(4));
+        items.add(person4);
+        return items;
+    }
+
+    private List<EntitySet> createEntityList(int itemsCount) {
+        List<EntitySet> items = new ArrayList<>();
+        EntitySet entity = null;
+        for (int i = 0; i < itemsCount; i++) {
+            entity = new EntitySet();
+            entity.setDescription("entity #" + i);
+            items.add(entity);
+        }
+        return items;
+    }
+
+    private <T extends GenericJson> List<String> getResultCheckFields(List<T> resultList, String checkField) {
+        List<String> result = new ArrayList<String>();
+        for (GenericJson item : resultList) {
+            result.add(item.get(checkField).toString());
+        }
+        return result;
+    }
+
+    private <T extends GenericJson> boolean checkIfItemsAtRightIndex(List<T> srcList, List<T> resultList,
+                                             int[] checkIndexes, String checkField, boolean checkErr, boolean checkIsBackendItem) throws AssertionError {
+
+        assertNotNull(srcList);
+        assertNotNull(resultList);
+
+        boolean result = true;
+        GenericJson srcItem;
+        GenericJson item;
+        int curIdx = 0;
+        List<String> resultFields = getResultCheckFields(resultList, checkField);
+        for (int idx : checkIndexes) {
+            item = resultList.get(curIdx);
+            if (checkErr) {
+                result &= item == null;
+            } else {
+                srcItem = srcList.get(idx);
+                result &= item != null;
+                if (checkIsBackendItem) {
+                    result &= isBackendItem(item);
+                }
+                String srcField = srcItem.get(checkField).toString();
+                result &= resultFields.contains(srcField);
```

Comment:
I used this logic because entities order can be different sometimes.
Then first i check if `checkField` contains in result array and get list of values of this field: